disagmoe/frontend/engine.py

```diff
@@ -257,16 +343,34 @@ def _pack_flash_attn_metadata(
             self,
             meta_c: AttentionBatchMetadata,
             meta_py: AttentionBatchMetadata,
-            decode_seq_lens: List[int]
+            decode_seq_lens: List[int],
+            mocking: bool = False,
         ) -> FlashAttentionMetadata:
         
         num_tokens = meta_py.num_decode_tokens + meta_py.num_prefill_tokens
         num_seqs = meta_py.num_prefill_seqs + meta_py.num_decode_tokens
         
         # 1. prepare block table
-        block_table_1d = self.block_mgr.prepare_block_table(meta_c, decode_seq_lens)
-        block_table_cuda = block_table_1d[ : -num_tokens].view(num_tokens, -1)
-        slot_mapping_cuda = block_table_1d[-num_seqs : ]
+        if not mocking:
+            block_table_1d = self.block_mgr.prepare_block_table(meta_c, decode_seq_lens)
+        else:
+            # mocking=True when _warmup_attn
+            block_table_1d = torch.zeros(
+                (num_tokens + num_seqs * MAX_SEQ_LEN // self.cache_config.block_size, ), 
+                dtype=torch.int32).to("cuda")
+        
+        if self.model_config.enable_cuda_graph:
+            self.static_slot_mapping[0: num_seqs].copy_(
+                block_table_1d[-num_seqs : ].to(torch.int64))
+            _max_num_blocks = (block_table_1d.numel() - num_seqs) // num_tokens
+            self.static_block_table[0: num_tokens, 0: _max_num_blocks].copy_(
```

Comment:
We can still use static block table as a 1D tensor, while passing it to executor with "viewing" as 2D tensor